cylc/flow/cfgspec/globalcfg.py

```diff
@@ -498,14 +498,61 @@
                     [[Platform_A]]
                         install target = localhost
             ''')
+
         with Conf('localhost', meta=Platform):
             Conf('hosts', VDR.V_STRING_LIST, ['localhost'])
 
     # Platform Groups
     with Conf('platform groups'):
         with Conf('<group>'):
             Conf('platforms', VDR.V_STRING_LIST)
-
+    # Symlink Dirs
+    with Conf('symlink dirs'):
+        with Conf('<install target>'):
+            Conf('run', VDR.V_STRING, None, desc="""
+                Specifies the directory where the workflow run directories are
+                created. If specified, the workflow run directory will be
+                created in <run dir>/cylc-run/<workflow-name> and a symbolic
+                link will be created from $HOME/cylc-run/<workflow-name>.
+                If not specified the workflow run directory will be created in
+                $HOME/cylc-run/<workflow-name>.
+                All the workflow files and the .service directory get installed
+                into this directory.
+            """)
+            Conf('log', VDR.V_STRING, None, desc="""
+                Specifies the directory where log directories are created. If
+                specified the workflow log directory will be created in
+                <log dir>/cylc-run/<workflow-name>/log and a symbolic link will
+                be created from $HOME/cylc-run/<workflow-name>/log. If not
+                specified the workflow log directory will be created in
+                $HOME/cylc-run/<workflow-name>/log.
+            """)
+            Conf('share', VDR.V_STRING, None, desc="""
+                Specifies the directory where share directories are created.
+                If specified the workflow share directory will be created in
+                <share dir>/cylc-run/<workflow-name>/share and a symbolic link
+                will be created from <$HOME/cylc-run/<workflow-name>/share. If
+                not specified the workflow share directory will be created in
+                $HOME/cylc-run/<workflow-name>/share.
+            """)
+            Conf('share/cycle', VDR.V_STRING, None, desc="""
```

Comment:
@oliver-sanders I'm getting namespace problems with this share/cycle when building the docs. Is there a way around this one or does it need changing to something like share-cycle?

Reply:
Ha, yes I suppose you would get a problem with that as `/` will severely jigger up URLs breaking the docs. Thanks for checking the docs!

Two options:

1) Rename as you suggest.
2) Do something fancy to cylc-sphinx-extensions to convert the `/` to something safe, but unique.

I think (2) might be a bit tricky.

How about trying `share>cycle`, `>` is also a nasty character for the web but it might get handled better. IDK.

Reply:
So, I have tried `share>cycle`, as you suspected, same namespace error.
Happy to have a go at 2. or go with `share-cycle`. I guess the problem with `share-cycle` is `cycle` isn't obviously a sub-directory of `share`. Hmm, do you think it is worth the time investment in 2?

Reply:
Can try at (2), if you can't fix it quickly-ish but it's probably not worth trying.

Would need to replace `/` with something benign, there are two likely places where this logic could go:

* https://github.com/cylc/cylc-sphinx-extensions/blob/85583e178c31b8d279e6df3553d0a9c4b7d93d60/cylc/sphinx_ext/cylc_lang/domains.py#L470-L493
* https://github.com/cylc/cylc-sphinx-extensions/blob/85583e178c31b8d279e6df3553d0a9c4b7d93d60/cylc/sphinx_ext/cylc_lang/domains.py#L341-L347

If you fancy a go try one or both of these.

I think since the cylc domain was written there is now some minimal Sphinx documentation for "custom domains" which might provide a pointer or two.